quarkchain/cluster/shard_state.py

```diff
@@ -1281,6 +1286,13 @@ def add_cross_shard_tx_list_by_minor_block_hash(
         - it is a neighor of current shard following our routing rule
         """
         self.db.put_minor_block_xshard_tx_list(h, tx_list)
+        for tx in tx_list.tx_list:
```

Comment:
as mentioned before, this function will also be called by the source shard, which we shouldn't do pub

Reply:
oh just saw your reply. please investigate why only the target shards run this code path

Reply:
Will do.